ethereum/trie/src/main/java/org/hyperledger/besu/ethereum/trie/TrieIterator.java

```diff
@@ -40,10 +41,15 @@ public TrieIterator(final LeafHandler<V> leafHandler, final boolean unload) {
   @Override
   public Node<V> visit(final ExtensionNode<V> node, final Bytes searchPath) {
     Bytes remainingPath = searchPath;
+    final Bytes extensionPath;
+    final Bytes ignoreRemainingPath;
     if (state == State.SEARCHING) {
-      final Bytes extensionPath = node.getPath();
-      final int commonPathLength = extensionPath.commonPrefixLength(searchPath);
-      remainingPath = searchPath.slice(commonPathLength);
+      extensionPath = node.getPath();
+      ignoreRemainingPath = searchPath.slice(0, Math.min(searchPath.size(), extensionPath.size()));
```

Comment:
nitpick: commonPrefixPath makes more sense to me than ignoreRemainingPath as a var name

Reply:
modified